modules/vm/templates/entrypoint.sh.tpl

```diff
@@ -11,8 +11,82 @@ until ping -c 1 google.com &> /dev/null; do
   sleep 5
 done
 
-# TODO: Find/create/mount volumes
-# https://learn.microsoft.com/en-us/azure/virtual-machine-scale-sets/tutorial-use-disks-cli
+# Find/create/attach volumes
+
+instanceHostname=\'$(hostname)\'
+subscriptionID=$(az account show --query "id" --output tsv)
+resourceGroup=$(az vmss list --query "[0].resourceGroup" --output tsv)
+vmssName=$(az vmss list --query "[0].name" --output tsv)
+instanceID=$(az vmss list-instances --resource-group $resourceGroup --name $vmssName --query "[?contains(osProfile.computerName, ${instanceHostname})].instanceId" --output tsv)
+zoneID=$(az vmss list-instances --resource-group $resourceGroup --name $vmssName --query "[?contains(osProfile.computerName, ${instanceHostname})].zones" --output tsv)
+regionID=$(az vmss list-instances --resource-group $resourceGroup --name $vmssName --query "[?contains(osProfile.computerName, ${instanceHostname})].location" --output tsv)
+# TODO replace with external variables
+tier="P40"
+lun=2 # device will be mounted as sdb
+graphdb_device="/dev/sdb"
+diskSizeGB=128
+
+# Define the disk name based on the hostname
+diskName="Disk_${vmssName}_${instanceID}"
+
+# Check if the device sdb exists
+if [ ! -b /dev/sdb ]; then
+    echo "Device /dev/sdb not found."
+
+    # Check if the managed disk is already attached to this instance
+    if [[ $(az vmss list-instances --resource-group $resourceGroup --name $vmssName --query "[?instanceId==\"$instanceID\"].disks[].name" --output tsv) == "$diskName" ]]; then
```

Comment:
When is it possible to be attached but not exist as a device?

Reply:
if the attach command fails for any reason

Reply:
Completely redid the logic